frontend/javascripts/oxalis/controller/scene_controller.ts

```diff
@@ -36,10 +36,53 @@ import window from "libs/window";
 import { setSceneController } from "oxalis/controller/scene_controller_provider";
 import { getSegmentColorAsHSLA } from "oxalis/model/accessors/volumetracing_accessor";
 import { mergeVertices } from "libs/BufferGeometryUtils";
+import { getTDViewportLOD } from "oxalis/model/accessors/view_mode_accessor";
 import { getPlaneScalingFactor } from "oxalis/model/accessors/view_mode_accessor";
+import { NO_LOD_MESH_INDEX } from "oxalis/model/sagas/isosurface_saga";
 
 const CUBE_COLOR = 0x999999;
 
+class CustomLOD extends THREE.LOD {
+  noLODGroup: THREE.Group;
+  lodLevelCount: number;
+  constructor() {
+    super();
+    this.lodLevelCount = 0;
+    this.noLODGroup = new THREE.Group();
+    this.add(this.noLODGroup);
+  }
+
+  update(_camera: any) {
+    const levels = this.levels;
+
+    const visibleIndex = getTDViewportLOD(Store.getState());
+    // Keep level 0 always visible as this includes all meshes that do not have different LODs.
+    for (let i = 0; i < this.levels.length; i++) {
+      levels[i].object.visible = i === visibleIndex;
```

Comment:
what happens if lod 0 wasn't loaded yet, but lod 1 would be available (likely, because lod 1 is smaller in byte size)? then, this code will only show lod 0 (which is empty) and not lod 1, correct?
one could probably check whether the current lod is empty (via `children`), but I suspect this wouldn't work on a per-chunk basis, right?

Reply:
> what happens if lod 0 wasn't loaded yet, but lod 1 would be available (likely, because lod 1 is smaller in byte size)? then, this code will only show lod 0 (which is empty) and not lod 1, correct?

Yes, that is why the code I wrote tries to prioritize loading the current LOD / triggers the backend calls for these chunks & LOD first.

> one could probably check whether the current lod is empty (via children),

Yeah, and then show a different LOD which is available / not empty? Sounds possible.

> , but I suspect this wouldn't work on a per-chunk basis, right?

As far as I understood it no. The reason would be that the chunks have a different size depending on the LOD and therefore making hard to find a different LOD and the matching chunks that would only cover the not yet fetched / received mesh chunks :/
If I am wrong with my assumption, please correct me :)

Reply:
Ok, then let's keep it as is for this iteration :)